lib/kubernetes-deploy/kubernetes_resource/custom_resource_definition.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class CustomResourceDefinition < KubernetesResource
+    TIMEOUT = 30.seconds
+
+    def deploy_succeeded?
+      names_accepted_status == "True"
+    end
+
+    def deploy_failed?
+      names_accepted_status == "False"
+    end
+
+    def timeout_message
+      UNUSUAL_FAILURE_MESSAGE
+    end
+
+    def group_version_kind
+      group = @definition.dig("spec", "group")
+      version = @definition.dig("spec", "version")
+      "#{group}/#{version}/#{kind}"
+    end
+
+    def kind
+      @definition.dig("spec", "names", "kind")
+    end
+
+    def prunable?
+      json_annotation = @definition.dig("metadata", "annotations", "kubernetes-deploy.shopify.io/metadata")
```

Comment:
What are your thoughts (@stefanmb too) on one big annotation vs. many smaller ones? Or an inbetween option that groups related items (e.g. predeploy/dependencies, success/failure/timeout)?

Reply:
I'd prefer to use smaller ones (e.x. `kubernetes-deploy.shopify.io/prunable` & kubernetes-deploy.shopify.io/predeploy`) that don't require us to store json strings.